raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -985,6 +986,8 @@ private boolean handleFetchResponse(
                 Records records = (Records) partitionResponse.recordSet();
                 if (records.sizeInBytes() > 0) {
                     LogAppendInfo info = log.appendAsFollower(records);
+                    log.flush();
```

Comment:
nit: maybe we can wrap the flushing and hwm updating logic in a `flushFollowerLog` as well.

Reply:
Makes sense.

Reply:
I considered this, but it's a little more awkward since the high watermark is coming from the request. However, I pushed an update which factors out an `appendAsFollower` call which is nearly symmetric with the existing `appendAsLeader`.